### ISSUE TITLE
[Trainer.py] tr_loss in trainer with distributed training

### ISSUE THREAD

**Issue by logoutlzh** [huggingface/transformers#12220]
## Environment info

- `transformers` version: 4.4, not very sure
- Platform: linux
- Python version: 3.7
- PyTorch version (GPU?): 1.7
- Tensorflow version (GPU?): //
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: Distributed training with single node with multi-gpu

### Who can help

@sgugger 

## Information

Model I am using (Bert, XLNet ...): Bart

The problem arises when using: trainer.py

## To reproduce

1. python -m torch.distributed.launch --nproc_per_node=2 xxx
2. observe the tr_loss

## Expected behavior

I'm not sure if it's a bug or a misunderstanding. In `trainer.py`, the `tr_loss` printed in the distributed training is the loss caused in rank = 0. Do we need to reduce the `tr_loss`?

![image](https://user-images.githubusercontent.com/23735761/122360089-b4c14900-cf88-11eb-9d2d-1e8a5b724a17.png)

**Comment by sgugger** [comment]
Since it's averaged over all the training mini-batches, it should be a good representation of the real training loss. I'd personally avoid any complexity and add a new reduce operation here, since a user can always evaluate on the training set to get the "real" training loss if they absolutely need to. Does that make sense?

**Comment by logoutlzh** [comment]
Thank you for your reply. I got it.
